apisix/balancer.lua

```diff
@@ -194,7 +195,20 @@ end
 -- 2. each time we need to retry upstream
 local function pick_server(route, ctx)
     core.log.info("route: ", core.json.delay_encode(route, true))
-    core.log.info("ctx: ", core.json.delay_encode(ctx, true))
+    local redacted_ctx = core.table.deepcopy(ctx)
+    for name, conf in pairs(ctx.var._ctx.consumer.plugins) do
+        local plugin = require("apisix.plugins."..name)
+        local schema
+        if plugin.type == "auth" then
+            schema = plugin.consumer_schema
+        else
+            schema = plugin.schema
+        end
+        redacted_ctx.var._ctx.consumer.plugins[name] = redact_encrypted(conf, schema)
+        local redacted_auth = redact_encrypted(redacted_ctx.var._ctx.consumer.auth_conf, schema)
+        redacted_ctx.var._ctx.consumer.auth_conf = redacted_auth
+    end
```

Comment:
this code snippet repeated multiple times, we should put it into a util function to reuse it.